lib/temp.js

```diff
@@ -103,7 +103,14 @@ function deleteDirOnExit(dirPath) {
 function attachExitListener() {
   if (!tracking) return false;
   if (!exitListenerAttached) {
-    process.addListener('exit', cleanupSync);
+    process.addListener('exit', function() {
+        try {
+            cleanupSync();
+        } catch(err) {
+            console.error("Fail to clean temporary files on exit : ", err);
```

Comment:
This should not throw an error. It should be a warning. Temp file removal is not high priority or considered an error for other applications.

```suggestion
            console.warn("Fail to clean temporary files on exit : ", err);
```

Reply:
I agree aminya. Should only warn, pulling this down and resolving.